sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/config/KeyVaultSecretProvider.java

```diff
@@ -8,10 +8,10 @@
 public interface KeyVaultSecretProvider {
```

Comment:
We still need this?

Reply:
This is for resolving key vault references without connecting to key vault. It has no "client", it is instead of.

Reply:
But internally we will build the KV clients?

Reply:
So the way this works is before we check if a client even exists we pass the KV Reference to the KeyVaultSecretProvider, if it returns any non-Null value than that is returned as the secret. If nothing is returned or this isn't configured only then do we create a client or check if one has already been made.